---
title: "Analyze selective constraints using polymorphism for young genes"
author: "Bin He"
date: "11/2/2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## background

This is work by Shao Yi in Zhang Yong's lab at IOZ of CAS. They have built a database cataloging primate-specific and human-specific genes. They compiled gene dating estimates from several sources and demonstrated that different methods each have their unique strengths and weaknesses and that when used together can enhance the confidence on the final estimate. They also took advantage of the newly curated dataset to perform a number of gene-level and genome-level analyses on the patterns of selective constraints, both for providing evidence for the young genes being functionally constrained and to give insight to the evolutionary forces behind the gain and loss of them.

## goal

In their analysis, they want to estimate the gene-level selective constraint using polymorphism data. The goal of this analysis is to test the feasibility of this approach. In particular, I was worried that given the typical level of nucleotide diversity in human, they are not going to get enough number of segregating variants to make any statistical test informative.

## data

Yong downloaded csv files for three example genes and sent to me.

## analysis

```{r load_data}
library(data.table)
raw <- list()
raw$myh16 <- fread("../data/MYH16.csv")
raw$glud1 <- fread("../data/GLUD1.csv")
raw$glud2 <- fread("../data/GLUD2.csv")
```

```{r SFS}
attach(raw)
## MYH16
myh16[,table(`Allele Count European (Non-Finnish)`)]
```
## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
